src/interrupt.c

```diff
@@ -600,6 +600,14 @@ void init_interrupts()
     }
 }
 
+/**
+ * @brief Initialize the interrupt controller
+ */
+void init_interrupts()
+{
+    __init_interrupts();
+}
```

Comment:
__init_interrupts is idempotent, but since it's been already called by the constructor there's really no need to call it again, I would say. In fact, you could even change this to function to a static inline function in the header file that is empty and thus removed even if compiled.

Reply:
Because of -Werror, if we define a function and then never use it an error will generated. Perhaps we just make it an macro instead that evaluates to nothing?

Reply:
I don't see this problem in rsp.h for deprecated functions, but anyway you can use `__attribute__((used))` to disable that warning.

Reply:
I think I was missing the `inline` part, that seems to have fixed it. I pushed a new commit that should address all of your comments.